Source/Core/VideoCommon/PixelShaderGen.cpp

```diff
@@ -1397,52 +1411,63 @@ constexpr std::array<const char*, 4> tev_alpha_funclogic_table{
     " == "   // xnor
 };
 
-static void WriteAlphaTest(ShaderCode& out, const pixel_shader_uid_data* uid_data, APIType ApiType,
+static void WriteAlphaTest(ShaderCode& out, const pixel_shader_uid_data* uid_data, APIType api_type,
                            bool per_pixel_depth, bool use_dual_source)
 {
-  static constexpr std::array<const char*, 2> alpha_ref{I_ALPHA ".r", I_ALPHA ".g"};
+  static constexpr std::array<std::string_view, 2> alpha_ref{
+      I_ALPHA ".r",
+      I_ALPHA ".g",
+  };
+
+  const auto write_alpha_func = [&out](int index, std::string_view ref) {
+    const bool has_no_arguments = index == 0 || index == 7;
```

Comment:
Instead of hardcoding 7, maybe use `tev_alpha_funcs_table.size() -1` ?  Or possibly assert here that `tev_alpha_funcs_table.size() == 8` , just to ensure this changes if the array gets modified?